model/task_creation.go

```diff
@@ -0,0 +1,34 @@
+package model
+
+import (
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/distro"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"time"
+)
+
+// TaskCreationInfo contains the needed parameters to construct new builds and tasks for a given version.
+type TaskCreationInfo struct {
+	Version             *Version
+	Project             *Project
+	ProjectRef          *ProjectRef
+	buildVariant        *BuildVariant
+	build               *build.Build
```

Comment:
is there a reason some of these are uppercase versus lowercase? (If the lowercase ones have certain in-package use, can they be moved below and commented?)

Reply:
No - this is a leftover from me merging the new task creation struct (which was initially private) with the old `BuildCreateArgs` struct which is/was public. Fixed now, they are all public.